src/radical/saga/adaptors/base.py

```diff
@@ -51,43 +53,45 @@ def __init__ (self, adaptor_info, adaptor_options=None, expand_env=True):
     # if sanity_check() is commented out here, then we will only load adaptors
     # which implement the method themselves.
     #
-    def sanity_check (self) :
-        """ This method can be overloaded by adaptors to check runtime
-            conditions on adaptor load time.  The adaptor should raise an
-            exception if it will not be able to function properly in the given
-            environment, e.g. due to missing dependencies etc.
-        """
-        raise BadParameter ("Adaptor %s does not implement sanity_check()"  \
-                         % self._name)
+    def sanity_check(self):
+        '''
+        This method can be overloaded by adaptors to check runtime conditions on
+        adaptor load time.  The adaptor should raise an exception if it will not
+        be able to function properly in the given environment, e.g. due to
+        missing dependencies etc.
+        '''
+
+        pass
```

Comment:
The doc string is not in sync with the code. It says that an exception is raised, but the method just passes.

Reply:
It is in sync ;-)  This is the base class' method - it _can_ be overloaded to raise an exception on check errors.  In fact, it did not adhere to the description before the change...